src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const char of actions) {
+    for (const key in char) {
```

Comment:
you don't need this loop - you have preset properties - so check them as you do below (you may also use `switch`)